core/testing/src/main/java/io/github/droidkaigi/confsched2023/testing/robot/BookmarkScreenRobot.kt

```diff
@@ -47,6 +60,36 @@ class BookmarkScreenRobot @Inject constructor(
             .captureRoboImage()
     }
 
+    fun clickBookmarkFilterChipAll() {
+        composeTestRule
+            .onNode(hasTestTag(BookmarkFilterChipAllTestTag))
+            .performClick()
+    }
+
+    fun clickBookmarkFilterChipDay1() {
+        composeTestRule
+            .onNode(hasTestTag(BookmarkFilterChipDay1TestTag))
+            .performClick()
+    }
+
+    fun clickBookmarkFilterChipDay2() {
+        composeTestRule
+            .onNode(hasTestTag(BookmarkFilterChipDay2TestTag))
+            .performClick()
+    }
+
+    fun clickBookmarkFilterChipDay3() {
+        composeTestRule
+            .onNode(hasTestTag(BookmarkFilterChipDay3TestTag))
+            .performClick()
+    }
+
+    fun toggleFavorite() = runTest(testDispatcher) {
+        SessionsAllResponse.fake().toTimetable().bookmarks.forEach {
+            userDataStore.toggleFavorite(it)
```

Comment:
Sorry, I have a question.
I tried to write a value using UserDataStore#toggleFavorite, but when I debugged it, it didn't seem to be written.
Could you please tell me what I should do?

Reply:
I think we can add suspend here. Could you check it?
<img width="610" alt="image" src="https://github.com/DroidKaigi/conference-app-2023/assets/1386930/bc4e704d-4a7a-43ef-803a-6e5da7d0980d">

Reply:
Thank you very much. However, when I ran the test, I got an error.

```
Detected use of different schedulers. If you need to use several test coroutine dispatchers, create one TestCoroutineScheduler and pass it to each of them.
```

The data used was incorrect and has been fixed in e83a729.

Reply:
May I ask if you use `suspend fun toggleFavorites` and `block: suspend BookmarkScreenRobot.() -> Unit` both ?

Reply:
Yes, I used the following code:

```kotlin
class BookmarkScreenRobot @Inject constructor(
    private val testDispatcher: TestDispatcher,
) {
    ...
    operator fun invoke(
        block: BookmarkScreenRobot.() -> Unit,
    ) {
        runTestWithLogging(timeout = 30.seconds) {
            this@BookmarkScreenRobot.composeTestRule = robotTestRule.composeTestRule
            toggleFavorites()
            block()
        }
    }
    ...
    suspend fun toggleFavorites() {
        SessionsAllResponse.fake().toTimetable().timetableItems.forEach {
            userDataStore.toggleFavorite(it.id)
        }
    }
}
```